### ISSUE TITLE
multiple_inherent_impl false positive

### ISSUE THREAD

**Issue by bugadani** [rust-lang/rust-clippy#5772]
Hi!

I have some code that uses multiple impls to add methods to specific generic parameters.
The code in question is [here](https://github.com/bugadani/embedded-layout/blob/ccd8a9d0e5f17cf279b496091f1381d9ea91f0b2/src/layout/linear/mod.rs#L98) but the warning also contains the "problematic" bits.
Clippy doesn't like this when I enable `restrictions`

```
warning: Multiple implementations of this structure
   --> src\layout\linear\mod.rs:121:1
    |
121 | / impl LinearLayout<Vertical<horizontal::Left, Tight>, Guard> {
122 | |     /// Create a new, empty [`LinearLayout`] that places views top to bottom
123 | |     #[inline]
124 | |     pub fn vertical() -> Self {
...   |
129 | |     }
130 | | }
    | |_^
    |
note: the lint level is defined here
   --> src\lib.rs:122:5
    |
122 |     clippy::restriction,
    |     ^^^^^^^^^^^^^^^^^^^
    = note: `#[warn(clippy::multiple_inherent_impl)]` implied by `#[warn(clippy::restriction)]`
note: First implementation here
   --> src\layout\linear\mod.rs:110:1
    |
110 | / impl LinearLayout<Horizontal<vertical::Bottom, Tight>, Guard> {
111 | |     /// Create a new, empty [`LinearLayout`] that places views left to right
112 | |     #[inline]
113 | |     pub fn horizontal() -> Self {
...   |
118 | |     }
119 | | }
    | |_^
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#multiple_inherent_impl  
```

**Comment by flip1995** [comment]
The restriction group is not meant to be enabled completely. Recently there was a lint implemented for this: https://rust-lang.github.io/rust-clippy/master/index.html#blanket_clippy_restriction_lints

Anyway, this lint should take generic parameters into account.
